redbot/cogs/mod/kickban.py

```diff
@@ -82,6 +82,16 @@ class KickBanMixin(MixinMeta):
         elif not (0 <= days <= 7):
             return _("Invalid days. Must be between 0 and 7.")
 
+        toggle = await self.settings.guild(guild).toggle_dm()
+        if toggle:
+            if guild.me.top_role > user.top_role and user != guild.owner and author != user:
```

Comment:
Now that you moved it, you can remove the `if guild.me.top_role > user.top_role and user != guild.owner and author != user:` here as well.